src/java2yaml/Steps/RestoreDependency.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace Microsoft.Content.Build.Java2Yaml
+{
+    using System;
+    using System.IO;
+    using System.Threading.Tasks;
+    using System.Linq;
+
+    public class RestoreDependency : IStep
+    {
+        public RestoreDependency(string path)
+        {
+            RepositoryPath = path;
+        }
+        public string StepName
+        {
+            get { return "RestoreDependency"; }
+        }
+
+        public string RepositoryPath { get; }
+
+
+        public Task RunAsync(ConfigModel config)
+        {
+            return Task.Run(() =>
+            {
+                Guard.ArgumentNotNullOrEmpty(RepositoryPath, nameof(RepositoryPath));
+
+                var projectFiles = FileUtility.GetFilesByName(RepositoryPath, Constants.BuildTool.Maven);
+                var packagePath = Path.Combine(RepositoryPath, Constants.PackageFolder);
+
+                if (projectFiles.Count() == 0)
+                {
+                    ConsoleLogger.WriteLine(new LogEntry
+                    {
+                        Phase = StepName,
+                        Level = LogLevel.Error,
+                        Message = " The project file pom.xml cannot be found, exiting..."
+                    });
+
+                    Environment.Exit(1);

```

Comment:
You can use `throw` here. Outside has a catch to handle this. Feels not like a common usage to interrupt inside program.

Reply:
switch to `throw new System.Exception()`